src/proof_production/Contradiction.cpp

```diff
@@ -0,0 +1,38 @@
+/*********************                                                        */
+/*! \file Contradiction.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Omri Isac, Guy Katz
+ ** This file is part of the Marabou project.
+ ** Copyright (c) 2017-2022 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved. See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** [[ Add lengthier description here ]]
+ **/
+
+#include <cstddef>
+#include "Contradiction.h"
+
+Contradiction::Contradiction( unsigned var, double *upperBoundExplanation, double *lowerBoundExplanation )
+        :_var( var )
+        ,_upperBoundExplanation( upperBoundExplanation )
+        ,_lowerBoundExplanation( lowerBoundExplanation )
+{
+}
+
+Contradiction::~Contradiction()
+{
+    if ( _upperBoundExplanation )
```

Comment:
It seems that these arrays are allocated somewhere and deleted here. We generally try to have the same class in charge of creating and deleting stuff. Maybe make a local copy, here?